rotkehlchen/accounting/cost_basis/base.py

```diff
@@ -100,11 +116,15 @@ def __str__(self) -> str:
         )
 
 
-class BaseAcquisitionsOrder(metaclass=ABCMeta):
-    _acquisitions: deque[AssetAcquisitionEvent]
+class BaseCostBasisMethod(metaclass=ABCMeta):
+    # For FIFO, the time at the point of insertion is used
+    # For LIFO, the time at the point of insertion is used with a negation
+    # For HIFO, the amount of the acquisition event is used with a negation
+    # The reason for the negation is because of Python's `heapq` implementation
+    _acquisitions: list[Tuple[ACQUISITION_PRIORITY, AssetAcquisitionEvent]]
```

Comment:
1. Why have it here? You do realize this is never used, right? Just initialize the type inside init.
2. You are misuing a `Tuple` here. Turn `[Tuple[ACQUISITION_PRIORITY, AssetAcquisitionEvent]` into an actual named tuple with 2 attributes and make it much easier to access and most of all **readable**.

I am struggling to read the code you have written. `x[0][1]`, `x[1]0]`  .... wtf?
Use a named tuple to make it obvious what is happening.


You also really need to explain the priority here or wherever you define the named tuple. Seems to be a counter going up in fifo. Same in LiFo but negated, I guess for the order? And then in hifo the amount with negation, so the smallest amount used first? Or highest? All these need to be much better explained and at the appropriate place.

And that place is the priority attribute of the named tuple I explained you need to add